backend/domain/domainmodels.js

```diff
@@ -0,0 +1,119 @@
+class Room {
+    constructor() {
+
+        this.roomName = this.setRoomName(); // some function to initialise a roomID
+
+        // default values given.
+        this.gamestate = GameStateEnum["Day Time"];
```

Comment:
Might be worth having these enums in a different file, so we can export them and use them as constants! :D I think then we can avoid using String values like this, and avoid errors if we mispell them or something

Reply:
What would we replace the string types with?

Reply:
Oh dam, my bad nevermind ignore me. I think I just wanted to ensure that we wouldn't muck up with using string literals, but I just tried it out in VSC and it helps you type out the correct string literal. I forgot this isn't something like typescript...